metricbeat/module/jolokia/jmx/data.go

```diff
@@ -104,31 +121,53 @@ func eventMapping(entries []Entry, mapping AttributeMapping) ([]common.MapStr, e
 
 	for _, v := range entries {
 		hasWildcard := strings.Contains(v.Request.Mbean, "*")
-		for attribute, value := range v.Value {
-			if !hasWildcard {
-				err := parseResponseEntry(v.Request.Mbean, v.Request.Mbean, attribute, value, mbeanEvents, mapping)
-				if err != nil {
-					errs = append(errs, err)
+		if v.Value == nil || v.Request.Attribute == nil {
+			continue
+		}
+		attributeType := reflect.TypeOf(v.Request.Attribute).String()
+		valueType := reflect.TypeOf(v.Value).String()
```

Comment:
Could [type switching](https://tour.golang.org/methods/16) be used here instead of reflect?

Reply:
Yes! I tried type switching first but it looks real messy. But I'm gonna give it a try again and please let me know what you think with the change. I think it's actually better with type switching. More readable I guess. Thanks!